packages/yew/src/utils/mod.rs

```diff
@@ -23,6 +23,15 @@ impl<IN: Into<OUT>, OUT> From<IN> for NodeSeq<IN, OUT> {
     }
 }
 
+impl<IN: Into<OUT>, OUT> From<Option<IN>> for NodeSeq<IN, OUT> {
+    fn from(val: Option<IN>) -> Self {
+        Self(
+            val.map(|s| vec![s.into()]).unwrap_or_default(),
```

Comment:
I think you can actually iterate over options:
```suggestion
            val.into_iter().collect(),
```

Reply:
I tested this on godbolt.org, and the two versions produce different assembly, where the former is slightly smaller (llvm seems to optimize the case that `val = None` differently) than using `collect`, so I stuck with that  one.

Reply:
I don't know :sweat_smile: my experience is more in high level languages...

Generally the rule is: if the intention is not obvious in the code, then it should be documented. So I would tend to say yes.

Long story:

Personally I wouldn't optimize that way because of my lack of experience. I just can't reason with it, I have too many questions in my mind. For example, is the code in assembly better for x86 or for WASM? Could it be different? Will that change in future releases of Rust/LLVM? I'm not asking you like they are relevant questions, I really don't know if those questions even make sense at all. I'm not even sure how to measure/evaluate which assembly code is faster (in terms of cycles or something I guess). So yeah I'm a total noob sorry xD I just can't answer for sure.

But I think that's the good part of the Rust community, we come with very different backgrounds. I prefer the readability of the iterator version but I totally agree it's more important to have fast code. The only reason for me to be against faster code is if it's very difficult to maintain (like if you would do everything unsafe (extreme scenario)). But here it's simple in both cases, the later code is more elegant which is a weak advantage compared to speed and size.